File: ports/1.21.1-full/src/main/java/mods/eln/ghost/GhostManagerNbt.java
package mods.eln.ghost;

import mods.eln.Eln;
import net.minecraft.nbt.CompoundTag;
import net.minecraft.world.level.saveddata.SavedData;

public class GhostManagerNbt extends SavedData {
    public GhostManagerNbt(String par1Str) {
        super(par1Str);
    }

    @Override
    public boolean isDirty() {
        return true;
    }

    @Override
    public void readFromNBT(CompoundTag nbt) {
        Eln.ghostManager.loadFromNBT(nbt);
    }

    @Override
    public CompoundTag writeToNBT(CompoundTag nbt) {
        //Eln.ghostManager.saveToNbt(nbt, Integer.MIN_VALUE);
        return nbt;
    }
}

File: src/main/kotlin/mods/eln/ghost/GhostManagerNbt.kt
package mods.eln.ghost

import mods.eln.Eln
import net.minecraft.core.HolderLookup
import net.minecraft.nbt.CompoundTag
import net.minecraft.world.level.saveddata.SavedData

/** The world-saved hook that loads the ghost manager (1.7.10's WorldSavedData). */
class GhostManagerNbt : SavedData() {
    override fun isDirty(): Boolean {
        return true
    }

    override fun save(nbt: CompoundTag, registries: HolderLookup.Provider): CompoundTag = nbt

    companion object {
        @JvmField
        val FACTORY = Factory({ GhostManagerNbt() }, { nbt, _ -> GhostManagerNbt().also { Eln.ghostManager.loadFromNBT(nbt) } }, null)
    }
}
